app/src/main/kotlin/com/hippo/ehviewer/util/Atomics.kt

```diff
@@ -0,0 +1,28 @@
+package com.hippo.ehviewer.util
+
+import kotlin.concurrent.atomics.AtomicInt
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+inline fun AtomicInt.loop(action: (Int) -> Unit): Nothing {
+    contract { callsInPlace(action, InvocationKind.AT_LEAST_ONCE) }
+    do {
+        action(load())
+    } while (true)
+}
+
+inline fun <R> AtomicInt.update(function: (Int) -> Int, transform: (old: Int, new: Int) -> R): R {
```

Comment:
The transform parameter doesn't quite fit the function name; is it even necessary?